server/src/main/java/org/opensearch/action/search/SearchUtils.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.StepListener;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.transport.RemoteClusterService;
+
+import java.util.Set;
+import java.util.function.BiFunction;
+
+/**
+ * Helper class for common search functions
+ */
+public class SearchUtils {
+
+    public SearchUtils() {}
+
+    /**
+     * Get connection lookup listener for list of clusters passed
+     */
+    public static ActionListener<BiFunction<String, String, DiscoveryNode>> getConnectionLookupListener(
+        RemoteClusterService remoteClusterService,
+        ClusterState state,
+        Set<String> clusters
+    ) {
+        final StepListener<BiFunction<String, String, DiscoveryNode>> lookupListener = new StepListener<>();
+
+        if (clusters.isEmpty()) {
+            lookupListener.onResponse((cluster, nodeId) -> state.getNodes().get(nodeId));
```

Comment:
For my understanding what is this call supposed to do

Reply:
so this is a a predicate which will give the node info given the cluster and node id.

- if clusters is empty, we will just give the node info from the current cluster state
- otherwise we will use remote cluster service to collect nodes from all clusters mentioned

Reply:
since this is step listener, we will return the listener and the callers will do whenComplete() to perform action after the node lookup. This is technically a synchronous flow since caller will wait for whenComplete .

```
public method1 {
    Listener listener = getConnListener();
    listener.whenComplete(do something);
}
```